src/source/core/Registry.bs

```diff
@@ -70,33 +88,53 @@ namespace mc
             for each key in sectionData
               value = sectionData[key]
               key = lcase(`${key}`)
-              section.write(key, formatJson(value))
+
+              if not section.write(key, formatJson(value))
+                success = false ' flip success=false if there is any failure; subsequent successes do not negate failure
+                if not success
+                  m.log.error("Failed to write registry data for registry section '" + sectionName + "', key '" + key + "'")
```

Comment:
you can use template strings such as  `Failed to write registry data for registry section  ${sectionName } , key  ${key}`

Reply:
Ah, fantastic!
I wasn't sure if they were supported in the older version of bsc

Fixed all these 🙂